client/app/bundles/course/assessment/pages/AssessmentGenerate/utils.ts

```diff
@@ -37,111 +54,90 @@ export function extractQuestionPrototypeData(
-              ?.map((testCase) => ({
-                expression: testCase.expression,
-                expected: 'True',
-                hint: testCase.hint,
-              })) ?? [],
+          public: buildTestCases('public', response),
+          private: buildTestCases('private', response),
+          evaluation: buildTestCases('hidden', response),
         },
       },
     },
   };
 }
 
+type IndexedTestCase = Record<
+  number,
+  { expression: string; expected: string; hint: string }
+>;
+
 export const buildGenerateRequestPayload = (
   codaveriData: CodaveriGenerateFormData,
   questionData: QuestionPrototypeFormData,
 ): FormData => {
   const data = new FormData();
-  // TODO: Currently we are injecting the existing question data into the custom prompt directly.
-  // When Codaveri implements this as a feature, make sure to use the updated request model.
-  const fragments = [codaveriData.customPrompt];
+  const isDefaultQuestionFormData =
+    JSON.stringify(questionData) === JSON.stringify(defaultQuestionFormData);
+
+  const populateTestCases = (
+    fieldName: string,
+    testCases: MetadataTestCase[],
+    testCaseDict: IndexedTestCase,
+  ): void => {
+    testCases.forEach((testCase, index) => {
+      testCaseDict[index + 1] = {
+        expression: testCase.expression,
+        expected: testCase.expected,
+        hint: testCase.hint,
+      };
+    });
+
+    data.append(fieldName, JSON.stringify(testCaseDict));
+  };
+
+  data.append(
+    'is_default_question_form_data',
+    isDefaultQuestionFormData.toString(),
+  );
+
   if (questionData?.question?.title) {
-    fragments.push(`title is currently "${questionData?.question?.title}"`);
+    data.append('title', questionData.question.title);
   }
 
   if (questionData?.question?.description) {
-    fragments.push(
-      `description is currently"${questionData?.question?.description}"`,
-    );
+    data.append('description', questionData.question.description);
   }
 
   if (questionData?.testUi?.metadata?.solution) {
-    fragments.push(
-      `solution is currently"${questionData?.testUi?.metadata?.solution}"`,
-    );
+    data.append('solution', questionData.testUi.metadata.solution);
   }
 
   if (questionData?.testUi?.metadata?.submission) {
-    fragments.push(
-      `template is currently"${questionData?.testUi?.metadata?.submission}"`,
-    );
+    data.append('template', questionData.testUi.metadata.submission);
   }
 
-  if (
-    questionData?.testUi?.metadata?.testCases?.public &&
-    questionData?.testUi?.metadata?.testCases?.public.length > 0
-  ) {
-    const subfragments = [`The current public test cases are:`];
-    questionData?.testUi?.metadata?.testCases?.public.forEach(
-      (testCase, index) => {
-        fragments.push(
-          `${index + 1}. ${testCase.expression} is ${testCase.expected} (${testCase.hint})`,
-        );
-      },
+  if (questionData?.testUi?.metadata?.testCases?.public?.length > 0) {
+    populateTestCases(
```

Comment:
minor nit: Since `testCaseDict` technically only exists in `populateTestCases` we can consider simplifying further by doing something like:

```ts
  const stringifyTestCases = (
    testCases: MetadataTestCase[],
  ): string => {
    const testCaseDict = {} as IndexedTestCase;
    testCases.forEach((testCase, index) => {
      testCaseDict[index + 1] = {
        expression: testCase.expression,
        expected: testCase.expected,
        hint: testCase.hint,
      };
    });

    return JSON.stringify(testCaseDict);
  };

...

const publicTestCases = questionData?.testUi?.metadata?.testCases?.public;
if (publicTestCases?.length > 0) {
  data.append('public_test_cases', publicTestCases);
}
```

Reply:
done